x-pack/plugin/sql/sql-shared-proto/src/main/java/org/elasticsearch/xpack/sql/proto/ProtoUtils.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.sql.proto;
+
+import org.elasticsearch.common.xcontent.ToXContent;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.common.xcontent.json.JsonXContent;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Locale;
+
+public final class ProtoUtils {
+
+    private ProtoUtils() {
+
+    }
+
+    /**
+     * Parses a generic value from the XContent stream
+     */
+    public static Object parseFieldsValue(XContentParser parser) throws IOException {
+        XContentParser.Token token = parser.currentToken();
+        Object value = null;
+        if (token == XContentParser.Token.VALUE_STRING) {
+            //binary values will be parsed back and returned as base64 strings when reading from json and yaml
+            value = parser.text();
```

Comment:
`return parser.text()` instead?

Reply:
Or is this copied from somewhere?

Reply:
Ah, yes, `XContentParserUtils`.

I'm fine with creating this for now, but maybe you can move this method to `:libs;x-content` in a followup? That'd be awesome. But tricky because the exception type changes. It'd *probably* be ok because we've made sure that the new exception type reports well though.

Reply:
I was looking into it and discussed it with @dakrone. It's tricky. I had to modify the it to get it here. The one in XContentParselUtils is also handling binary data that it converts into lucene's BytesArray and throws an exception that builds on the top of server's ElasticsearchException hierarchy. Pulling it out would be quite complicated without major refactoring of all existing users.

Reply:
I think doing what you've got is fine. Could you open up an issue to discuss removing the duplication?